### PR TITLE
(HTCONDOR-1598)  Miscellany

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
This PR depends on the previous 10 PRs, and needs all of them merged to build cleanly.  The target branch generates the relevant diff, but we'll leave this PR in draft until it can be rebased into a buildable state.

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: bindings/python/htcondor2/_startd.py [hunk 1]
```diff
@@ -1 +1,92 @@
...
+        # We never actually use this for anything.
+        # self._version = location['CondorVersion']
+
+
+    # In version 1, check_expr and start_expr could also be `ExprTree`s.
+    def drainJobs(self,
+      drain_type : DrainType = DrainType.Graceful,
+      on_completion : CompletionType = CompletionType.Nothing,
+      check_expr : str = "True",
+      start_expr : str = "False",
```
Comment (johnkn) [inline bindings/python/htcondor2/_startd.py:42]: these should be nullptr rather than defaulted to expression strings.

Comment (Todd-L-Miller) [reply bindings/python/htcondor2/_startd.py:42]: Did you mean `None`?

Comment (Todd-L-Miller) [reply bindings/python/htcondor2/_startd.py:42]: (or that the C code should send up empty strings if `check_expr` or `start_expr` were not specified?)

**Review by johnkn — APPROVED** [review]

File: bindings/python/htcondor2/_startd.py (hunk 1 above)
Comment (Todd-L-Miller) [reply bindings/python/htcondor2/_startd.py:42]: Punted to HTCONDOR-2163.

**Pull request merged**